src/main/scala/uk/gov/nationalarchives/dp/client/Client.scala

```diff
@@ -0,0 +1,139 @@
+  ): Client[F, S] = new Client[F, S] {
+    val dataProcessor: DataProcessor[F] = DataProcessor[F]()
+
+    implicit class EitherUtils[T](e: Either[String, T]) {
+      def bodyLift: F[T] = me.fromTry(e.left.map(err => new RuntimeException(err)).toTry)
+    }
+
+    private def getAuthenticationToken(authDetails: AuthDetails): F[String] = {
+      val response = basicRequest
+        .body(Map("username" -> authDetails.userName, "password" -> authDetails.password))
+        .post(uri"$apiBaseUrl/api/accesstoken/login")
+        .response(asJson[Token])
+        .send(backend)
+      me.flatMap(response)(res => me.fromTry(res.body.map(b => b.token).toTry))
+    }
+
+    private def xmlForUrl(url: String, token: String): F[Elem] = {
+      val request = basicRequest
+        .get(uri"$url")
+        .headers(Map("Preservica-Access-Token" -> token))
+        .response(asXml)
+
+      me.flatMap(backend.send(request))(_.body.bodyLift)
+    }
+
+    private def updatedEntities(
+        url: Option[String],
+        token: String,
+        allEntities: Seq[Entity]
+    ): F[Seq[Entity]] = {
+      if (url.isEmpty) {
+        me.pure(allEntities)
+      } else {
+        for {
+          elem <- xmlForUrl(url.get, token)
+          nextPage <- dataProcessor.nextPage(elem)
+          updateEntities <- dataProcessor.updatedEntities(elem)
+          all <- updatedEntities(nextPage, token, allEntities ++ updateEntities)
+        } yield all
+      }
+    }
+
+    override def getBitstreamInfo(
+        contentId: UUID,
+        authDetails: AuthDetails
+    ): F[Seq[BitStreamInfo]] = {
+      for {
+        token <- getAuthenticationToken(authDetails)
+        contentEntity <- xmlForUrl(s"$apiBaseUrl/api/entity/content-objects/$contentId", token)
+        generationUrl <- dataProcessor.generationUrlFromEntity(contentEntity)
+        generationInfo <- xmlForUrl(generationUrl, token)
+        allGenerationUrls <- dataProcessor.allGenerationUrls(generationInfo)
+        allGenerationElements <- allGenerationUrls.map(url => xmlForUrl(url, token)).sequence
+        allBitstreams <- dataProcessor.allBitstreamInfo(allGenerationElements)
+      } yield allBitstreams
+    }
+
+    override def metadataForEntityUrl(url: String, auth: AuthDetails): F[Seq[Elem]] = {
+      for {
+        token <- getAuthenticationToken(auth)
+        entityInfo <- xmlForUrl(url, token)
+        fragmentUrls <- dataProcessor.fragmentUrls(entityInfo)
+        fragmentResponse <- fragmentUrls.map(url => xmlForUrl(url, token)).sequence
+        fragments <- dataProcessor.fragments(fragmentResponse)
+      } yield {
+        fragments.map(XML.loadString)
+      }
+    }
+
+    override def entitiesUpdatedSince(
+        dateTime: ZonedDateTime,
+        authDetails: AuthDetails
+    ): F[Seq[Entity]] = {
+      val dateString = dateTime.format(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX"))
+      val queryParams = Map("date" -> dateString, "max" -> "100", "start" -> "0")
```

Comment:
The number for max is one of those things that we'll have to tweak when it's running, it's hard to tell now what's best so that's a good a choice as any.

We only need to set the step on the first call. The next page url in the initial response has `?start=100` in it.